db/migrate/20201222120101_add_position_to_spree_assemblies_parts.rb

```diff
@@ -4,8 +4,8 @@ class AddPositionToSpreeAssembliesParts < SolidusSupport::Migration[4.2]
   def up
     add_column :spree_assemblies_parts, :position, :integer, limit: 2
 
-    product_ids = Spree::AssembliesPart.pluck(:assembly_id)
-    Spree::Product.where(id: product_ids).find_each do |product|
+    product_ids = Spree::AssembliesPart.distinct(:assembly_id).pluck(:assembly_id)
+    Spree::Product.with_deleted.where(id: product_ids).find_each do |product|
```

Comment:
`.with_deleted` is deprecated on the latest Solidus versions, it should be better using `.with_discarded` now.